app/(candidate)/onboarding/[slug]/[step]/components/districts/DistrictPicker.js

```diff
@@ -0,0 +1,58 @@
+'use client'
+import React, { useState } from 'react'
+import BlackButtonClient from '@shared/buttons/BlackButtonClient'
+import DistrictTypeAutocomplete from './DistrictTypeAutocomplete'
+import DistrictNameAutocomplete from './DistrictNameAutocomplete'
+
+export default function DistrictPicker({
+  state,
+  electionYear,        // number, derived from electionDate
+  buttonText = 'Save',
+  onSubmit,
+  className = '',
+}) {
+  const [type, setType]   = useState(null)
+  const [name, setName]   = useState(null)
+  const [busy, setBusy]   = useState(false)
+
+  const canSubmit = !!type && !!name && !busy
+
+  const handleClick = async () => {
+    if (!canSubmit) return
+    setBusy(true)
+    try {
+      await onSubmit(type, name)
+    } finally {
+      setBusy(false)
+    }
+  }
```

Comment:
No need for `try-finally` here.  The async operation (i.e. `fetch`) down-flow should have a `try-catch`